lib/vizkit/widgets/task_inspector/task_inspector.rb

```diff
@@ -44,6 +44,36 @@ def init
             @filter.on_hide do
                 treeView.disconnect
             end
+
+            lineEdit.connect(SIGNAL('returnPressed()')) do
+                proxyModel.setFilterFixedString(lineEdit.text)
+            end
+
+            @proxyModel = Qt::SortFilterProxyModel.new(self)
+            @proxyModel.setSourceModel(@model)
+            @proxyModel.setDynamicSortFilter(true)
+            @proxyModel.setFilterKeyColumn(0)
+
+            def @proxyModel.filterAcceptsRow(source_row, source_parent)
```

Comment:
Better create a subclass of `SortFilterProxyModel`. The singleton method mechanism used for TreeView is (I believe ...) because of the usage of Qt designer.